app/src/main/java/com/example/parstagram/fragments/GalleryFragment.java

```diff
@@ -0,0 +1,174 @@
+                String description = etDescription.getText().toString();
+                if (description.isEmpty()) {
+                    Toast.makeText(getContext(), "Description cannot be empty", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                if (photoFile == null || ivPostImage.getDrawable() == null) {
+                    Toast.makeText(getContext(), "There is no image!", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                ParseUser currentUser = ParseUser.getCurrentUser();
+                savePost(description, currentUser, photoFile);
+                pb.setVisibility(ProgressBar.INVISIBLE);
+            }
+        });
+
+        btnSetProfile.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                pb.setVisibility(ProgressBar.VISIBLE);
+                if (photoFile == null || ivPostImage.getDrawable() == null) {
+                    Toast.makeText(getContext(), "There is no image!", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                ParseFile parseFile = new ParseFile(photoFile);
+                parseFile.saveInBackground();
+                ParseUser user = ParseUser.getCurrentUser();
+                user.put("profilePicture", parseFile);
+                user.saveInBackground(new SaveCallback() {
+                    @Override
+                    public void done(ParseException e) {
+                        if (e == null) {
+                            Toast.makeText(getContext(), "Successfully set profile picture!", Toast.LENGTH_SHORT).show();
+                        } else {
+                            e.printStackTrace();
+                        }
+                    }
+                });
+                pb.setVisibility(ProgressBar.INVISIBLE);
+            }
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RESULT_LOAD_IMAGE && null != data) {
+            final Uri uri = data.getData();
+            Bitmap bitmap = null;
+            try {
+                bitmap = MediaStore.Images.Media.getBitmap(getContext().getContentResolver(), uri);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            ByteArrayOutputStream stream = new ByteArrayOutputStream();
+            // Compress image to lower quality scale 1 - 100
+            bitmap.compress(Bitmap.CompressFormat.JPEG, 100, stream);
+            byte[] image = stream.toByteArray();
+
+            // Create the ParseFile
+            ParseFile file  = new ParseFile("picture_1.jpeg", image);
+            // Upload the image into Parse Cloud
+            ParseUser user = ParseUser.getCurrentUser();
+            user.put("profilePicture",file);
+
```

Comment:
Those file, path and key names should be stored in final variables initialized at the very beginning of your class. Otherwise it's gonna get complicated when you have to change something